pkg/api/types.go

```diff
@@ -180,6 +180,112 @@ type VolumeSource struct {
 	GitRepo *GitRepo `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretSource `json:"secret"`
+
+	// TODO (per thockin) https://github.com/GoogleCloudPlatform/kubernetes/pull/4055#discussion_r24865963
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
+
+	NFSMount *NFSMount `json:"nfsMount"`
```

Comment:
I will need to remove AWSEBS and NFS mount from this implementation.  They started here so I can account for the various AccessModes, but those plugins can be implemented separately and I'll handle the merge conflict when they land.